pkg/dcgm/field_values.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dcgm
+
+/*
+#include "dcgm_agent.h"
+#include "dcgm_structs.h"
+#include "field_values_cb.h"
+extern int go_dcgmFieldValueEntityEnumeration(dcgm_field_entity_group_t entityGroupId,
+            dcgm_field_eid_t entityId,
+            dcgmFieldValue_v1 *values,
+            int numValues,
+            void *userData);
+*/
+import "C"
+import (
+	"fmt"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+// FieldValueEntity represents a field value entity
+type FieldValueEntity struct {
+	ts    int64
```

Comment:
Just wondering if it's worth to keep ts as Time directly, making the right conversion here, instead of leaving that to a user.